src/mcmc/mcmc_misc.jl

```diff
@@ -174,44 +199,135 @@ end
-    iota = median(img0.nelec_per_nmgy)
-    gal_frac_devs = [gal_frac_dev, 1 - gal_frac_dev]
-    #XiXi = DeterministicVI.get_bvn_cov(ce.gal_axis_ratio, ce.gal_angle, ce.gal_radius_px)
-    XiXi = Model.get_bvn_cov(gal_axis_ratio, gal_angle, gal_radius_px)
+                                gal_scale::Float64,
+                                pixels::Matrix{Float64};
+                                offset::Array{Float64, 1}=[0., 0.],
+                                flux::Float64=1.)
+    # write the unit-flux scaled, correctly offset galaxy shape + 
+    # psf convolution
+    e_devs = [gal_frac_dev, 1 - gal_frac_dev]
+    XiXi = Model.get_bvn_cov(gal_axis_ratio, gal_angle, gal_scale)
+    for i in 1:2
+        for gproto in galaxy_prototypes[i]
+            for k in 1:length(psf)
+                the_mean = SVector{2}(WCS.world_to_pix(wcs, pos) - offset) +
+                           psf[k].xiBar
+                the_cov = psf[k].tauBar + gproto.nuBar * XiXi
+                intensity = flux * iota * psf[k].alphaBar * e_devs[i] * gproto.etaBar
+                write_gaussian(the_mean, the_cov, intensity, pixels)
+            end
+        end
+    end
+end
 
+function write_galaxy_unit_flux_pixel(px_pos::Array{Float64, 1},
+                                      psf::Array{Model.PsfComponent,1},
+                                      iota::Float64,
+                                      gal_frac_dev::Float64,
+                                      gal_axis_ratio::Float64,
+                                      gal_angle::Float64,
+                                      gal_scale::Float64,
+                                      pixels::Matrix{Float64};
+                                      flux::Float64=1.)
+    # write the unit-flux scaled, correctly offset galaxy shape + 
+    # psf convolution
+    e_devs = [gal_frac_dev, 1 - gal_frac_dev]
+    XiXi = Model.get_bvn_cov(gal_axis_ratio, gal_angle, gal_scale)
     for i in 1:2
         for gproto in galaxy_prototypes[i]
-            for k in 1:length(img0.psf)
-                the_mean = SVector{2}(WCS.world_to_pix(img0.wcs, pos)) +
-                           img0.psf[k].xiBar
-                the_cov = img0.psf[k].tauBar + gproto.nuBar * XiXi
-                intensity = iota * img0.psf[k].alphaBar * gal_frac_devs[i] *
-                    gproto.etaBar
+            for k in 1:length(psf)
+                the_mean = px_pos + psf[k].xiBar
+                the_cov = psf[k].tauBar + gproto.nuBar * XiXi
+                intensity = flux * iota * psf[k].alphaBar * e_devs[i] * gproto.etaBar
                 write_gaussian(the_mean, the_cov, intensity, pixels)
             end
         end
     end
 end
 
+
+"""
+Generate a model image on a patch, according to that image/patch psf
+"""
+function render_patch(img::Image, patch::SkyPatch, n_bodies::Vector{CatalogEntry})
+    # sky noise an gain
+    epsilon = img.sky[1, 1]
+    iota    = Float64(median(img.nelec_per_nmgy))
```

Comment:
Also potentially a problem because `nelec_per_nmgy` varies per row, both in the real images and the synthetic images (from Synthetic.jl).